source/features/pull-request-timeline-filter.tsx

```diff
@@ -0,0 +1,260 @@
+		</div>
+	);
+
+	form.append(actionButtons);
+
+	// This works on github enterprise - form is already preloaded
+	select('form', detailsDialog)?.remove();
+	// This works on normal github. Normally form is loaded in place of `include-fragment` after we open details dialog.
+	select('include-fragment', detailsDialog)?.remove();
+
+	detailsDialog.append(form);
+}
+
+async function tryClickLoadMore(item: HTMLElement): Promise<any> {
+	if (CurrentSettings.AutoLoadHidden) {
+		// Just after loading page when user clicks that element he is redirected to some limbo. It happens because github javascript did not kick in yet.
+		// To mitigate that we always give 1 second for javascript to load and notice this element so clicking it will be handled properly.
+		await sleep(1);
+		item.click();
+	}
+}
+
+function processTimelineItem(item: HTMLElement): void {
+	const pr = select('.js-comment[id^=pullrequestreview]', item);
+	const commitGroup = select('.js-commit-group', item);
+	const normalComment = select('.js-comment-container', item);
+
+	if (pr) {
+		processPR(item);
+	} else if (commitGroup) {
+		applyDisplay(item, CurrentSettings.HideCommits);
+	} else if (normalComment) {
+		applyDisplay(item, CurrentSettings.hideNormalComment);
+	} else {
+		applyDisplay(item, CurrentSettings.HideOthers);
+	}
+}
+
+function processPR(item: HTMLElement): void {
+	let hasVisibleElement = false;
+
+	for (const threadContainer of select.all('.js-resolvable-timeline-thread-container', item)) {
+		const commentContainer = select('.inline-comment-form-container', threadContainer);
+
+		if (threadContainer.getAttribute('data-resolved') === 'true') {
+			applyDisplay(threadContainer, CurrentSettings.HideResolved);
+		} else if (commentContainer === null) {
+			// There is 1 special case here when github shows you a comment that was added to previous comment thread but it does not show whether it is resolved or not resolved comment.
+			// It's kinda tricky to know what to do with this so it is marked as normal comment for meantime.
+			// We are just checking here if user is able to comment inside that timeline thread, if not then it means we have this special situation that was just described.
+			applyDisplay(threadContainer, CurrentSettings.hideNormalComment);
+		} else {
+			applyDisplay(threadContainer, CurrentSettings.HideUnresolved);
+		}
+
+		// We need to hide whole thread group if we have hidden all comments inside.
+		hasVisibleElement = hasVisibleElement || threadContainer.style.display === '';
+	}
+
+	applyDisplay(item, !hasVisibleElement);
+}
+
+async function init(): Promise<any> {
+	await addTimelineItemsFilter();
+
+	// There are some cases when github will remove this filter. In that case we need to add it again.
+	// Example: Editing comment will make timeline filter to disappear.
+	observe(`#${timelineFiltersSelectorId}`, {
+		async remove() {
```

Comment:
If you use `add: addTimelineItemsFilter` instead you can avoid having to call the function at all, even line 231 can be dropped. I *think* that this is how we manage the sidebar updates, you can double-check other sidebar features

Reply:
I do not know what you mean. Could you write short example or show me file where it is used?

Reply:
I left a new review for this